app/lib/katello/validators/hostgroup_kickstart_repository_validator.rb

```diff
@@ -19,12 +19,25 @@ def validate(hostgroup)
               elsif hostgroup.architecture.blank?
                 _("Please select an architecture before assigning a kickstart repository")
               elsif hostgroup.operatingsystem.kickstart_repos(hostgroup).none? { |repo| repo[:id] == hostgroup.kickstart_repository_id }
-                _("The selected kickstart repository is not part of the assigned content view, lifecycle environment,
-                  content source, operating system, and architecture")
+                if (equivalent = find_equivalent_kickstart(hostgroup))
```

Comment:
Instead of simply erroring out, we attempt to find a kickstart repo with a matching label in cases where perhaps the cv or env changed. the `find_equivalent_kickstart` method should probably be moved - open to ideas there.